.github/workflows/run_smoke_test.yaml

```diff
@@ -0,0 +1,67 @@
+on: 
+  workflow_dispatch:
+  pull_request:
+    types: [opened, reopened]
+    branches:
+      - "main"
+  pull_request_review:
+      types: [submitted]  
+  push:
+    branches:
+      - "main"
+
+name: Smoke Test
+
+jobs:
+  Run-Smoke-Test:
+    runs-on: windows-latest
+    env:
+      SCUBA_GITHUB_AUTOMATION_CREDS: ${{ secrets.SCUBA_GITHUB_AUTOMATION_CREDS }}
+    defaults:
+          run:
+            shell: powershell 
+    permissions:
+      contents: read 
+    steps:
+    - name: Checkout repo code
+      uses: actions/checkout@v3
+    - name: Execute ScubaGear and Check Outputs
+      run: |
+        . Testing/Functional/SmokeTest/SmokeTestUtils.ps1
+        ./AllowBasicAuthentication.ps1 -RunAsAdmin
+
+        ##### Install all the dependencies
+        Install-SmokeTestExternalDependencies
+
+        $AUTOMATION_CREDS = $env:SCUBA_GITHUB_AUTOMATION_CREDS | ConvertFrom-Json
```

Comment:
Please elaborate.  I don't understand the comment.

Reply:
```suggestion
        # ScubaGear currently requires the provisioning of a certificate for using a ServicePrinicpal, rather than
        # using Workload Identity Federation, which would ordinarily be preferred for calling Microsoft APIs from
        # GitHub actions.
        $AUTOMATION_CREDS = $env:SCUBA_GITHUB_AUTOMATION_CREDS | ConvertFrom-Json
```